### PR TITLE
Kernel: Improve the aarch64 kernel source files disk layout

### PR DESCRIPTION
**Pull request by jamesmintram**
As discussed in Discord: https://discord.com/channels/830522505605283862/897113928168013855/959956066337177640

The aarch64 folder has been getting a little full, with files which are mixed between rpi specific, prekernel and kernel.

The goal here is to move files into subdirectories which make more sense. There has been `prekernel` and `rpi`  folders added.

### CODE REVIEW

**Review by nico — APPROVED** [review]
LGTM

(One of the bots is unhappy, not sure of that's related.)

**Review by linusg — CHANGES_REQUESTED** [review]
The codebase almost exclusively uses `PascalCase` directory names, including `Kernel/Prekernel/*`. Let's please stick to that and not introduce lowercase `prekernel` as well :)

(`aarch64`/`x86` are outliers)

**Comment by jamesmintram** [comment]
> The codebase almost exclusively uses `PascalCase` directory names, including `Kernel/Prekernel/*`. Let's please stick to that and not introduce lowercase `prekernel` as well :)
> 
> (`aarch64`/`x86` are outliers)

Done!

File: Kernel/CMakeLists.txt [hunk 1]
```diff
@@ -416,12 +429,12 @@ else()
         Arch/aarch64/boot.S
         Arch/aarch64/Aarch64_asm_utils.S
         Arch/aarch64/vector_table.S
-
         Arch/aarch64/SmapDisabler.cpp
         Arch/aarch64/ScopedCritical.cpp
-        MiniStdLib.cpp
-        Prekernel/UBSanitizer.cpp
+
```
Comment (linusg) [inline Kernel/CMakeLists.txt:434]: Trailing whitespace here

**Review by linusg — CHANGES_REQUESTED** [review]

File: Kernel/CMakeLists.txt [hunk 2]
```diff
@@ -389,22 +389,35 @@ if (NOT "${SERENITY_ARCH}" STREQUAL "aarch64")
         ${CRYPTO_SOURCES}
     )
 else()
+    set(PREKERNEL_SOURCES
+        Arch/aarch64/prekernel/PrekernelMMU.cpp
+        Arch/aarch64/prekernel/PrekernelExceptions.cpp
+        Arch/aarch64/prekernel/PrekernelCommon.cpp
+        Arch/aarch64/prekernel/boot.S
+        Arch/aarch64/prekernel/Aarch64_asm_utils.S
```
Comment (linusg) [inline Kernel/CMakeLists.txt:397]: These still need to be updated. Is your FS case insensitive by any chance? Otherwise this shouldn't even build...

Comment (jamesmintram) [reply Kernel/CMakeLists.txt:397]: Yes it is! Lemme login to my Linux box and do this...

**Pull request merged**